languages/rust/concepts/structs/about.md

```diff
@@ -1 +1,94 @@
-TODO: add information on structs concept
+# Structs
+
+Structs collect a fixed, heterogenous set of data into a single unit. Structs in Rust have these properties:
+
+- Every struct type has a particular name.
+- Every member of a struct is called a **field**.
+- Every field of a struct has a particular name.
+- Visibility is tracked at the field level: each field may be independently public, private, or other[^1].
+
+# Tuples etc.
+
+Structs are very much like **Tuples**. In fact, the only difference between structs and tuples has to do with what is anonymous. The general form goes like this:
+
+- Structs are named, and have named fields:
+
+  ```rust
+  struct Foo{
+      bar: Bar,
+      bat: Bat,
+  }
+  ```
+
+  Fields are accessed by name:
+
+  ```rust
+  let bar = foo.bar;
+  ```
+
+- Tuple structs are named, and have anonymous fields:
+
+  ```rust
+  Foo(Bar, Bat)
+  ```
+
+  Fields are accessed by unpacking or by numeric index:
+
+  ```rust
+  // unpacking
+  let (bar, bat) = foo;
+  ```
+
+  ```rust
+  // numeric index
+  let bar = foo.0;
+  let bat = foo.1;
+  ```
+
+- Tuples are anonymous, and have anonymous fields:
+
+  ```rust
+  (Bar, Bat)
+  ```
+
+  As in tuple structs, fields are accessed by unpacking or by numeric index.
+
+- Anonymous structs are, as the name suggests, anonymous, but with named fields. You may be familiar with them from other languages, such as go or Java. They do not exist in Rust; it is considered idiomatic to use a tuple instead.
+
+# Zero Sized Types
+
+Structs can have as many or as few fields as is required. The implication is that it is legal for a struct to have zero fields. A struct with zero fields is called a **Zero-Sized Type** and has some special handling by the compiler: zero-sized types (often abbreviated as **ZSTs**) never consume stack or heap space, even when you might expect them to. For example, `Box<T>` doesn't allocate any heap memory if `T` is a ZST.
+
+You can use standard or tuple struct syntax when writing a ZST:
+
+```rust
+// declaration
+struct Foo {}
+// instantiation
+let foo = Foo {}
+```
+
+```rust
+// declaration
+struct Foo();
+// instantiation
+let foo = Foo();
+```
+
+There's also a shorthand for ZSTS:
```

Comment:
```suggestion
There's also a shorthand for ZSTs:
```

A few lines above, it says "often abbreviated as ZSTs" so I think the second S is lowercase (as it is not part of the initialism, just indicates plural)